tests/unittests/analysis/test_regret.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+"""Tests :meth:`orion.analysis.regret()"""
+import pandas as pd
+
+
+from orion.analysis.regret import regret
+
+
+def test_accept_dict():
+    """Tests dictionary parameter is supported"""
+    data = {
+        'id': ['a', 'b', 'c', 'd'],
+        'objective': [0.1, 0.2, 0.3, 0.5]
+    }
+
+    result = regret(data)
+
+    assert type(result) is pd.DataFrame
+
+
+def test_accept_empty():
+    """Tests an empty dataframe is returned if you give an empty dataframe"""
+    empty_frame = pd.DataFrame()
+    result = regret(empty_frame)
+
+    assert result.empty
+    assert result.equals(empty_frame)
+
+    empty_frame = pd.DataFrame(columns=['id', 'objective'])
+    result = regret(empty_frame)
+
+    assert result.empty
+    assert 'best' not in result.columns
+    assert 'best_id' not in result.columns
+    assert result.equals(empty_frame)
+
+
+def test_parameter_not_modified():
+    """Tests the original dataframe is not modified"""
+    data = pd.DataFrame(data={
+        'id': ['a', 'b', 'c', 'd'],
+        'objective': [0.1, 0.2, 0.3, 0.5]
+    })
+
+    result = regret(data)
+
+    assert len(data.columns) == 2
+    assert len(result.columns) == 4
+
+
+def test_default_column_names():
+    """Tests default column names"""
+    data = pd.DataFrame(data={
+        'id': ['a', 'b', 'c', 'd'],
+        'objective': [0.1, 0.2, 0.3, 0.5]
+    })
+
+    result = regret(data)
+
+    assert 'best' in result.columns
+    assert 'best_id' in result.columns
+
+
+def test_input_column_names():
+    """Tests custom column names"""
+    data = pd.DataFrame(data={
+        'id': ['a', 'b', 'c', 'd'],
+        'objective': [0.1, 0.2, 0.3, 0.5]
+    })
+
+    result = regret(data, names=('something', 'else'))
+
+    assert 'something' in result.columns
+    assert 'else' in result.columns
+
+
+def test_regret_sequential():
+    """Tests that cumulative best objectives are linked to their respective ids"""
+    data = pd.DataFrame(data={
+        'id': ['a', 'b', 'c', 'd'],
+        'objective': [10, 12, 8, 9],
```

Comment:
We should test also in case where there is many equals. I believe current implementation would override the ids. I would be in favor of keeping previous id in case of equality.

Reply:
Fixed in 7798278